src/solver/variable/include/antares/solver/variable/economy/STSbyGroup.h

```diff
@@ -0,0 +1,379 @@
+        {
+            unsigned int groupNumber = groupToNumbers_[cluster.properties.groupName];
+            // Injection
+            pValuesForTheCurrentYear[numSpace][3 * groupNumber][state.hourInTheYear]
+                += state.hourlyResults->ShortTermStorage[state.hourInTheWeek].injection[clusterIndex];
+
+            // Withdrawal
+            pValuesForTheCurrentYear[numSpace][3 * groupNumber + 1][state.hourInTheYear]
+                += state.hourlyResults->ShortTermStorage[state.hourInTheWeek].withdrawal[clusterIndex];
+
+            // Levels
+            pValuesForTheCurrentYear[numSpace][3 * groupNumber + 2][state.hourInTheYear]
+                += state.hourlyResults->ShortTermStorage[state.hourInTheWeek].level[clusterIndex];
+
+            clusterIndex++;
+        }
+
+        // Next variable
+        NextType::hourForEachArea(state, numSpace);
+    }
+
+    inline void buildDigest(SurveyResults& results, int digestLevel, int dataLevel) const
+    {
+        // Ask to build the digest to the next variable
+        NextType::buildDigest(results, digestLevel, dataLevel);
+    }
+
+    Antares::Memory::Stored<double>::ConstReturnType retrieveRawHourlyValuesForCurrentYear(
+      unsigned int column,
+      unsigned int numSpace) const
+    {
+        return pValuesForTheCurrentYear[numSpace][column].hour;
+    }
+
+    inline uint64_t memoryUsage() const
+    {
+        uint64_t r = (sizeof(IntermediateValues) * nbGroups_ + IntermediateValues::MemoryUsage())
+                         * pNbYearsParallel;
+        r += sizeof(double) * nbGroups_ * maxHoursInAYear * pNbYearsParallel;
+        r += AncestorType::memoryUsage();
+        return r;
+    }
+
+    std::string caption(unsigned int column) const
+    {
+        std::string groupName = groupNames_[column / 3];
+        std::string variableKind = VAR_POSSIBLE_KINDS[column % 3];
+        return groupName + "_" + variableKind;
+    }
+
+    std::string unit(unsigned int column) const
+    {
+        return (column % 3 == 2) ? "MWh" : "MW"; // Level in MWh, others in "MW"
+    }
+
+    void localBuildAnnualSurveyReport(SurveyResults& results,
+                                      int fileLevel,
+                                      int precision,
+                                      unsigned int numSpace) const
+    {
+        // Initializing external pointer on current variable non applicable status
+        results.isCurrentVarNA = AncestorType::isNonApplicable;
+
+        if (!AncestorType::isPrinted[0])
+            return;
+
+        for (unsigned int column = 0; column < nbColumns_; column++)
+        {
+            results.variableCaption = caption(column);
+            results.variableUnit = unit(column);
+            pValuesForTheCurrentYear[numSpace][column]
+                .template buildAnnualSurveyReport<VCardType>(results, fileLevel, precision);
+        }
+    }
+
+    void buildSurveyReport(SurveyResults& results,
```

Comment:
**buildSurveyReport** is called when building the columns in **synthesis** results.
If **buildSurveyReport** method is defined in an output var class, it is called.
If not, defined, it is the **IVariable** class's  **buildSurveyReport** that is called.
About this function, 2 remarks : 
- We **really** **really** should rename it, because it has an horrible name.
- take a look at **IVariable**'s version of this function : it calls **info.h**'s **BuildSurveyReport** function, and this latter calls **setClusterCaption** which is such an horrible function contains many **if**s. My advice would be to define a ****buildSurveyReport** in variable classes for which **categoryFileLevel** is **Category::de_res** or **Category::de_sts**